clap_complete_nushell/tests/completion.rs

```diff
@@ -0,0 +1,328 @@
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use nu_cli::NuCompleter;
+use nu_command::create_default_context;
+use nu_parser::parse;
+use nu_protocol::{
+    engine::{EngineState, Stack, StateWorkingSet},
+    Value,
+};
+
+use reedline::{Completer, Suggestion};
+
+const SEP: char = std::path::MAIN_SEPARATOR;
+
+// creates a new engine with the current path into the completions fixtures folder
+pub fn new_engine() -> (PathBuf, EngineState, Stack) {
```

Comment:
The `pub` keyword is not needed here, or maybe we could move the helper functions to common.rs which makes the code cleaner.